File: java/scf-bp-component/src/main/java/org/mydotey/scf/bp/component/labeled/DcSetting.java
package org.mydotey.scf.bp.component.labeled;

/**
 * @author koqizhao
 *
 * Jun 19, 2018
 */
public class DcSetting implements Cloneable {

    public static final String APP_KEY = "app";
    public static final String DC_KEY = "dc";

    private String key;
    private String app;
    private String dc;
    private String value;

    public DcSetting() {

    }

    public DcSetting(String key, String app, String dc, String value) {
        super();
        this.key = key;
        this.app = app;
        this.dc = dc;
        this.value = value;
    }

    public String getKey() {
        return key;
    }

    public String getApp() {
        return app;
    }

    public String getDc() {
        return dc;
    }

    public String getValue() {
        return value;
    }

    @Override
    public int hashCode() {
        final int prime = 31;
        int result = 1;
        result = prime * result + ((app == null) ? 0 : app.hashCode());
        result = prime * result + ((dc == null) ? 0 : dc.hashCode());
        result = prime * result + ((key == null) ? 0 : key.hashCode());
        return result;
    }

    @Override
    public boolean equals(Object obj) {
        if (this == obj)
            return true;
        if (obj == null)
            return false;
        if (getClass() != obj.getClass())
            return false;
        DcSetting other = (DcSetting) obj;
        if (app == null) {
            if (other.app != null)
                return false;
        } else if (!app.equals(other.app))
            return false;
        if (dc == null) {
            if (other.dc != null)
                return false;
        } else if (!dc.equals(other.dc))
            return false;
        if (key == null) {
            if (other.key != null)
                return false;
        } else if (!key.equals(other.key)) {
            return false;
        }

        return true;
    }

    @Override
    public String toString() {
        return String.format("%s { key: %s, app: %s, dc: %s, value: %s }", getClass().getSimpleName(), key, app, dc,
                value);
    }

    @Override
    public DcSetting clone() {
        try {
            return (DcSetting) super.clone();
        } catch (CloneNotSupportedException e) {
            e.printStackTrace();
            return null;
        }
    }

}

File: dotnet/src/MyDotey.SCF.BP.Component/Labeled/DcSetting.cs
using System;

namespace MyDotey.SCF.BP.Component.Labeled
{
    /**
     * @author koqizhao
     *
     * Jun 19, 2018
     */
    public class DcSetting : ICloneable
    {
        public static string APP_KEY = "App";
        public static string DC_KEY = "DC";

        public string Key { get; set; }
        public string App { get; set; }
        public string Dc { get; set; }
        public string Value { get; set; }

        public DcSetting()
        {

        }

        public DcSetting(string key, string app, string dc, string value)
        {
            Key = key;
            App = app;
            Dc = dc;
            Value = value;
        }

        public override int GetHashCode()
        {
            int prime = 31;
            int result = 1;
            result = prime * result + ((App == null) ? 0 : App.GetHashCode());
            result = prime * result + ((Dc == null) ? 0 : Dc.GetHashCode());
            result = prime * result + ((Key == null) ? 0 : Key.GetHashCode());
            return result;
        }

        public override bool Equals(object obj)
        {
            if (object.ReferenceEquals(this, obj))
                return true;
            if (obj == null)
                return false;
            if (GetType() != obj.GetType())
                return false;
            DcSetting other = (DcSetting)obj;
            if (App == null)
            {
                if (other.App != null)
                    return false;
            }
            else if (!App.Equals(other.App))
                return false;
            if (Dc == null)
            {
                if (other.Dc != null)
                    return false;
            }
            else if (!Dc.Equals(other.Dc))
                return false;
            if (Key == null)
            {
                if (other.Key != null)
                    return false;
            }
            else if (!Key.Equals(other.Key))
            {
                return false;
            }

            return true;
        }

        public override string ToString()
        {
            return string.Format("{0} {{ Key: {1}, App: {2}, Dc: {3}, Value: {4} }}", GetType().Name, Key, App, Dc,
                    Value);
        }

        public virtual object Clone()
        {
            return MemberwiseClone();
        }
    }
}